File: Day-11-of-30-Days/Solution.java
// Day 11 of 30 Days 👾

import java.time.*;
import java.util.*;
public class Solution {
    public static int[] calculateAge(String dob) {

        LocalDate birthDate = LocalDate.parse(dob, java.time.format.DateTimeFormatter.ofPattern("dd-MM-yyy"));
        LocalDate currentDate = LocalDate.now();

        Period period = Period.between(birthDate, currentDate);

        int days = period.getDays();
        int months = period.getMonths();
        int years = period.getYears();

        return new int[] { days, months, years };
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(calculateAge("10-06-2023"))); // Output Format : [10, 9, 0]
        System.out.println(Arrays.toString(calculateAge("12-11-1993"))); // Output Format: [8, 4, 30]
    }
}

File: Day-11-of-30-Days/Solution.py
# Day 11 of 30 Days 👾

from datetime import datetime

def calculate_age(dob):
    birth_date = datetime.strptime(dob, "%d-%m-%Y").date()
    current_date = datetime.now().date()

    delta = current_date - birth_date

    years = delta.days // 365
    months = (delta.days % 365) // 30
    days = (delta.days % 365) % 30

    return [days, months, years]

# Example 1
print(calculate_age("10-07-2023"))  # Output Format: [14, 8, 0]

# Example 2
print(calculate_age("12-11-1993"))  # Output Format: [16, 4, 30]